### PR TITLE
[production/RRFS.v1] The final smoke update PR to RRFS.v1

### PR DESCRIPTION
**Pull request by haiqinli**
## Commit Queue Requirements:

- [ ] Fill out all sections of this template.
- [ ] All sub component pull requests have been reviewed by their code managers.
- [ ] Run the full Intel+GNU RT suite (compared to current baselines) on either Hera/Derecho/Hercules
- [ ] Commit 'test_changes.list' from previous step
---
### Commit Message:

```
* UFSWM - 
 
  * FV3 - 
    * ccpp-physics - 


### Priority:
<!--
Please provide the priority you would prefer this pull request to have.

* High: Time-sensitive project.

Please delete the ones that are not applicable
-->

* High: For the RRFS.v1.


## Git Tracking
### UFSWM:

### Sub component Pull Requests:
<!--
Please provide a list of sub-components involved with this pull request.
Please provide links to the sub-component pull requests as shown below.
Please delete what is not needed.

* FV3: NOAA-EMC/fv3atm#841
  * ccpp-physics: ufs-community/ccpp-physics#211


### UFSWM Blocking Dependencies:


---
## Changes
### Regression Test Changes (Please commit test_changes.list):
<!--
Please let us know if this PR creates new baselines, changes baselines or not.
Please delete what is not needed.
Please make sure you have properly submitted test_changes.list
-->

* PR Updates/Changes Baselines.


### Input data Changes:
<!--
If there are any changes to input-data for a test, please provide information here.
Please delete what is not needed.
-->
* None.


### Library Changes/Upgrades:


  
---
<!-- STOP!!! THE FOLLOWING IS FOR CODE MANAGERS ONLY. PLEASE DO NOT FILL OUT -->
## Testing Log:
- RDHPCS
  - [ x] Hera
  - [ ] Orion
  - [ ] Hercules
  - [ ] Jet
  - [ ] Gaea
  - [ ] Derecho
- WCOSS2
  - [ ] Dogwood/Cactus
  - [ ] Acorn
- [ ] CI
- [ ] opnReqTest (complete task if unnecessary)

### CODE REVIEW

**Comment by jkbk2004** [comment]
@MatthewPyle-NOAA I am going to create new baseline on hera/hercules.

**Comment by MatthewPyle-NOAA** [comment]
@jkbk2004 I will do the same on WCOSS

**Comment by grantfirl** [comment]
@MatthewPyle-NOAA @jkbk2004 @haiqinli is on leave until next Tuesday.

**Comment by jkbk2004** [comment]
> @MatthewPyle-NOAA @jkbk2004 @haiqinli is on leave until next Tuesday.

@grantfirl thanks for the note!

**Comment by MatthewPyle-NOAA** [comment]
Not sure what the issue is, but seeing errors like this one in a few tests on WCOSS (such as conus13km_debug_intel)

```
forrtl: severe (19): invalid reference to variable in NAMELIST input, unit -5, file Internal Formatted NML Read, line -1, position 20
Image              PC                Routine            Line        Source
fv3.exe            000000000BD1336B  Unknown               Unknown  Unknown
fv3.exe            000000000BD4E779  Unknown               Unknown  Unknown
fv3.exe            000000000BD4C830  Unknown               Unknown  Unknown
fv3.exe            000000000646DBC1  gfs_typedefs_mp_c        4465  GFS_typedefs.F90
fv3.exe            0000000006220886  gfs_init_mp_gfs_i          68  GFS_init.F90
fv3.exe            00000000021F6405  atmos_model_mod_m         709  atmos_model.F90
fv3.exe            0000000001C242E4  module_fcst_grid_         787  module_fcst_grid_comp.F90
fv3.exe            0000000000E2A194  _ZN5ESMCI6FTable1        2167  ESMCI_FTable.C
fv3.exe            0000000000E2DDCA  ESMCI_FTableCallE         824  ESMCI_FTable.C
fv3.exe            000000000142E95B  _ZN5ESMCI3VMK5ent        1125  ESMCI_VMKernel.C
fv3.exe            0000000001417DD9  _ZN5ESMCI2VM5ente        1216  ESMCI_VM.C
fv3.exe            0000000000E2B5D7  c_esmc_ftablecall         981  ESMCI_FTable.C
fv3.exe            0000000000A0A710  esmf_compmod_mp_e        1223  ESMF_Comp.F90
fv3.exe            00000000008110B6  esmf_gridcompmod_        1412  ESMF_GridComp.F90
fv3.exe            0000000001BDB3A2  fv3atm_cap_mod_mp         393  fv3_cap.F90
fv3.exe            0000000000C79A4F  _ZNK5ESMCI13Metho         377  ESMCI_MethodTable.C
fv3.exe            0000000000C799CD  _ZN5ESMCI11Method         563  ESMCI_MethodTable.C
fv3.exe            0000000000C7A381  c_esmc_methodtabl         347  ESMCI_MethodTable.C
fv3.exe            0000000000B7496B  esmf_attachmethod        1280  ESMF_AttachMethods.F90
fv3.exe            0000000005081068  nuopc_modelbase_m         687  NUOPC_ModelBase.F90
fv3.exe            0000000000E2A194  _ZN5ESMCI6FTable1        2167  ESMCI_FTable.C
fv3.exe            0000000000E2DDCA  ESMCI_FTableCallE         824forrtl: severe (19): invalid reference to variable in NAMELIST input, unit -5, file Internal Formatted NML Read, line -1, position 20
```

**Comment by BrianCurtis-NOAA** [comment]
it doesn't like a variable in input.nml. Did you change any of them recently?

**Comment by MatthewPyle-NOAA** [comment]
@haiqinli Is it ready to test again?

**Comment by haiqinli** [comment]
> Not sure what the issue is, but seeing errors like this one in a few tests on WCOSS (such as conus13km_debug_intel)
> 
> ```
> forrtl: severe (19): invalid reference to variable in NAMELIST input, unit -5, file Internal Formatted NML Read, line -1, position 20
> Image              PC                Routine            Line        Source
> fv3.exe            000000000BD1336B  Unknown               Unknown  Unknown
> fv3.exe            000000000BD4E779  Unknown               Unknown  Unknown
> fv3.exe            000000000BD4C830  Unknown               Unknown  Unknown
> fv3.exe            000000000646DBC1  gfs_typedefs_mp_c        4465  GFS_typedefs.F90
> fv3.exe            0000000006220886  gfs_init_mp_gfs_i          68  GFS_init.F90
> fv3.exe            00000000021F6405  atmos_model_mod_m         709  atmos_model.F90
> fv3.exe            0000000001C242E4  module_fcst_grid_         787  module_fcst_grid_comp.F90
> fv3.exe            0000000000E2A194  _ZN5ESMCI6FTable1        2167  ESMCI_FTable.C
> fv3.exe            0000000000E2DDCA  ESMCI_FTableCallE         824  ESMCI_FTable.C
> fv3.exe            000000000142E95B  _ZN5ESMCI3VMK5ent        1125  ESMCI_VMKernel.C
> fv3.exe            0000000001417DD9  _ZN5ESMCI2VM5ente        1216  ESMCI_VM.C
> fv3.exe            0000000000E2B5D7  c_esmc_ftablecall         981  ESMCI_FTable.C
> fv3.exe            0000000000A0A710  esmf_compmod_mp_e        1223  ESMF_Comp.F90
> fv3.exe            00000000008110B6  esmf_gridcompmod_        1412  ESMF_GridComp.F90
> fv3.exe            0000000001BDB3A2  fv3atm_cap_mod_mp         393  fv3_cap.F90
> fv3.exe            0000000000C79A4F  _ZNK5ESMCI13Metho         377  ESMCI_MethodTable.C
> fv3.exe            0000000000C799CD  _ZN5ESMCI11Method         563  ESMCI_MethodTable.C
> fv3.exe            0000000000C7A381  c_esmc_methodtabl         347  ESMCI_MethodTable.C
> fv3.exe            0000000000B7496B  esmf_attachmethod        1280  ESMF_AttachMethods.F90
> fv3.exe            0000000005081068  nuopc_modelbase_m         687  NUOPC_ModelBase.F90
> fv3.exe            0000000000E2A194  _ZN5ESMCI6FTable1        2167  ESMCI_FTable.C
> fv3.exe            0000000000E2DDCA  ESMCI_FTableCallE         824forrtl: severe (19): invalid reference to variable in NAMELIST input, unit -5, file Internal Formatted NML Read, line -1, position 20
> ```

@MatthewPyle-NOAA Yes, there is a namelist change fo smoke_forecast to hwp_method. We can simply delete the smoke_forecast in the input.nml of RT cases, and it worked well after deleting the smoke_forecast in /scratch2/BMC/acomp/Haiqin.Li/stmp4/Haiqin.Li/FV3_RT/rt_3775248/conus13km_debug_intel.

**Comment by haiqinli** [comment]
> @haiqinli Is it ready to test again?

@MatthewPyle-NOAA Yes, it is ready to test again now. Thanks.

**Comment by jkbk2004** [comment]
> > Not sure what the issue is, but seeing errors like this one in a few tests on WCOSS (such as conus13km_debug_intel)
> > ```
> > forrtl: severe (19): invalid reference to variable in NAMELIST input, unit -5, file Internal Formatted NML Read, line -1, position 20
> > Image              PC                Routine            Line        Source
> > fv3.exe            000000000BD1336B  Unknown               Unknown  Unknown
> > fv3.exe            000000000BD4E779  Unknown               Unknown  Unknown
> > fv3.exe            000000000BD4C830  Unknown               Unknown  Unknown
> > fv3.exe            000000000646DBC1  gfs_typedefs_mp_c        4465  GFS_typedefs.F90
> > fv3.exe            0000000006220886  gfs_init_mp_gfs_i          68  GFS_init.F90
> > fv3.exe            00000000021F6405  atmos_model_mod_m         709  atmos_model.F90
> > fv3.exe            0000000001C242E4  module_fcst_grid_         787  module_fcst_grid_comp.F90
> > fv3.exe            0000000000E2A194  _ZN5ESMCI6FTable1        2167  ESMCI_FTable.C
> > fv3.exe            0000000000E2DDCA  ESMCI_FTableCallE         824  ESMCI_FTable.C
> > fv3.exe            000000000142E95B  _ZN5ESMCI3VMK5ent        1125  ESMCI_VMKernel.C
> > fv3.exe            0000000001417DD9  _ZN5ESMCI2VM5ente        1216  ESMCI_VM.C
> > fv3.exe            0000000000E2B5D7  c_esmc_ftablecall         981  ESMCI_FTable.C
> > fv3.exe            0000000000A0A710  esmf_compmod_mp_e        1223  ESMF_Comp.F90
> > fv3.exe            00000000008110B6  esmf_gridcompmod_        1412  ESMF_GridComp.F90
> > fv3.exe            0000000001BDB3A2  fv3atm_cap_mod_mp         393  fv3_cap.F90
> > fv3.exe            0000000000C79A4F  _ZNK5ESMCI13Metho         377  ESMCI_MethodTable.C
> > fv3.exe            0000000000C799CD  _ZN5ESMCI11Method         563  ESMCI_MethodTable.C
> > fv3.exe            0000000000C7A381  c_esmc_methodtabl         347  ESMCI_MethodTable.C
> > fv3.exe            0000000000B7496B  esmf_attachmethod        1280  ESMF_AttachMethods.F90
> > fv3.exe            0000000005081068  nuopc_modelbase_m         687  NUOPC_ModelBase.F90
> > fv3.exe            0000000000E2A194  _ZN5ESMCI6FTable1        2167  ESMCI_FTable.C
> > fv3.exe            0000000000E2DDCA  ESMCI_FTableCallE         824forrtl: severe (19): invalid reference to variable in NAMELIST input, unit -5, file Internal Formatted NML Read, line -1, position 20
> > ```
> 
> @MatthewPyle-NOAA Yes, there is a namelist change fo smoke_forecast to hwp_method. We can simply delete the smoke_forecast in the input.nml of RT cases, and it worked well after deleting the smoke_forecast in /scratch2/BMC/acomp/Haiqin.Li/stmp4/Haiqin.Li/FV3_RT/rt_3775248/conus13km_debug_intel.

@haiqinli @MatthewPyle-NOAA what about https://github.com/haiqinli/ufs-weather-model/blob/production/RRFS.v1-final/tests/parm/rrfs_conus13km_hrrr.nml.IN#L152 ? I mean to address the failed issue.

**Comment by haiqinli** [comment]
> > > Not sure what the issue is, but seeing errors like this one in a few tests on WCOSS (such as conus13km_debug_intel)
> > > ```
> > > forrtl: severe (19): invalid reference to variable in NAMELIST input, unit -5, file Internal Formatted NML Read, line -1, position 20
> > > Image              PC                Routine            Line        Source
> > > fv3.exe            000000000BD1336B  Unknown               Unknown  Unknown
> > > fv3.exe            000000000BD4E779  Unknown               Unknown  Unknown
> > > fv3.exe            000000000BD4C830  Unknown               Unknown  Unknown
> > > fv3.exe            000000000646DBC1  gfs_typedefs_mp_c        4465  GFS_typedefs.F90
> > > fv3.exe            0000000006220886  gfs_init_mp_gfs_i          68  GFS_init.F90
> > > fv3.exe            00000000021F6405  atmos_model_mod_m         709  atmos_model.F90
> > > fv3.exe            0000000001C242E4  module_fcst_grid_         787  module_fcst_grid_comp.F90
> > > fv3.exe            0000000000E2A194  _ZN5ESMCI6FTable1        2167  ESMCI_FTable.C
> > > fv3.exe            0000000000E2DDCA  ESMCI_FTableCallE         824  ESMCI_FTable.C
> > > fv3.exe            000000000142E95B  _ZN5ESMCI3VMK5ent        1125  ESMCI_VMKernel.C
> > > fv3.exe            0000000001417DD9  _ZN5ESMCI2VM5ente        1216  ESMCI_VM.C
> > > fv3.exe            0000000000E2B5D7  c_esmc_ftablecall         981  ESMCI_FTable.C
> > > fv3.exe            0000000000A0A710  esmf_compmod_mp_e        1223  ESMF_Comp.F90
> > > fv3.exe            00000000008110B6  esmf_gridcompmod_        1412  ESMF_GridComp.F90
> > > fv3.exe            0000000001BDB3A2  fv3atm_cap_mod_mp         393  fv3_cap.F90
> > > fv3.exe            0000000000C79A4F  _ZNK5ESMCI13Metho         377  ESMCI_MethodTable.C
> > > fv3.exe            0000000000C799CD  _ZN5ESMCI11Method         563  ESMCI_MethodTable.C
> > > fv3.exe            0000000000C7A381  c_esmc_methodtabl         347  ESMCI_MethodTable.C
> > > fv3.exe            0000000000B7496B  esmf_attachmethod        1280  ESMF_AttachMethods.F90
> > > fv3.exe            0000000005081068  nuopc_modelbase_m         687  NUOPC_ModelBase.F90
> > > fv3.exe            0000000000E2A194  _ZN5ESMCI6FTable1        2167  ESMCI_FTable.C
> > > fv3.exe            0000000000E2DDCA  ESMCI_FTableCallE         824forrtl: severe (19): invalid reference to variable in NAMELIST input, unit -5, file Internal Formatted NML Read, line -1, position 20
> > > ```
> > 
> > 
> > @MatthewPyle-NOAA Yes, there is a namelist change fo smoke_forecast to hwp_method. We can simply delete the smoke_forecast in the input.nml of RT cases, and it worked well after deleting the smoke_forecast in /scratch2/BMC/acomp/Haiqin.Li/stmp4/Haiqin.Li/FV3_RT/rt_3775248/conus13km_debug_intel.
> 
> @haiqinli @MatthewPyle-NOAA what about https://github.com/haiqinli/ufs-weather-model/blob/production/RRFS.v1-final/tests/parm/rrfs_conus13km_hrrr.nml.IN#L152 ? I mean to address the failed issue.

@jkbk2004 Yes, "smoke_forecast" in that file should be changed to "hwp_method". I will update it soon. Thanks.

**Comment by MatthewPyle-NOAA** [comment]
@haiqinli Did you have a chance to update it?

**Comment by haiqinli** [comment]
@MatthewPyle-NOAA Ravan will give me some smoke updates tomorrow, and I will update this PR tomorrow. Thanks.

**Comment by haiqinli** [comment]
@MatthewPyle-NOAA @grantfirl @jkbk2004 Per our discussion with Ming and Curtis, we will split this PR into two seperate PRs. The new PR only includes the FMS and minor RUC LSM deep soil temperature updates, and this new PR could be merged into RRFS.v1 easily. The smoke updates will be kept in the current PR. However, we need to do some retro runs to guarantee there will be very minor impacts on the meteorology forecast, and may take more time. Thanks.

**Comment by jkbk2004** [comment]
> @MatthewPyle-NOAA @grantfirl @jkbk2004 Per our discussion with Ming and Curtis, we will split this PR into two seperate PRs. The new PR only includes the FMS and minor RUC LSM deep soil temperature updates, and this new PR could be merged into RRFS.v1 easily. The smoke updates will be kept in the current PR. However, we need to do some retro runs to guarantee there will be very minor impacts on the meteorology forecast, and may take more time. Thanks.

@haiqinli thanks for the update!

**Comment by haiqinli** [comment]
@MatthewPyle-NOAA Should we move this PR (the smoke updates) forward? Thanks.

**Comment by MatthewPyle-NOAA** [comment]
@haiqinli If it has been confirmed that the impact on meteorological forecasts is small, I'd say sure.

**Comment by haiqinli** [comment]
> @haiqinli If it has been confirmed that the impact on meteorological forecasts is small, I'd say sure.

@MatthewPyle-NOAA 
t1
t2
t3
t4

We conducted sensitivity simulations with our smoke PR for the RRFS-A. We've used the input and boundary conditions from the real-time RRFS_A system in these simulations. The main goals of this PR are the following: 
- to fix a bug affecting diurnal cycle of agriculture fire emissions
- to add a new namelist option to scale the fire emissions (based on NESDIS's RAVE satellite fire dataset), when we switch to RAVEv2. The current operational RAVE data (v1.3) ingested into RRFS_A has a major bug, which leads to overestimation of fire emissions. We'll need to apply a scaling factor to the RAVE data, when we switch to ingesting RAVEV2. NOAA's current operational AQM model (UFS-CMAQ) is also using the RAVE data, and is going to update the AQM to ingest the RAVE v2 data sometime late summer.
- 
The figures are the averaged temperature and wind speed differences (averaged over the CONUS domain) between the simulations with the current RRFS_A code and our changes. As we expected, the differences are negligible. The reason for this is that the smoke impact on meteorology is local, in areas with dense smoke. Usually agricultural fires don't emit tons of smoke. ((This PR is revised to remove the microphysics changes).

**Comment by MatthewPyle-NOAA** [comment]
@haiqinli Can you clean up the conflicts so we can look to move this forward?

**Comment by haiqinli** [comment]
@MatthewPyle-NOAA Sure, I will sync the code with the authoritative repository and clean up the conflicts.

**Comment by MatthewPyle-NOAA** [comment]
@haiqinli Any idea when this will be ready to move forward?

**Comment by haiqinli** [comment]
@MatthewPyle-NOAA Yes, the PRs for ccpp-physics, fv3atm, and ufs-weather-model are updated, and are ready to go. Thanks.

**Comment by jkbk2004** [comment]
@haiqinli I see some conus13km cases crash. 
```
 27: forrtl: severe (19): invalid reference to variable in NAMELIST input, unit -5, file Internal Formatted NML Read, line -1, p
osition 20
 27: Image              PC                Routine            Line        Source
 27: fv3.exe            000000000399ADC8  Unknown               Unknown  Unknown
 27: fv3.exe            00000000039D6ED7  Unknown               Unknown  Unknown
 27: fv3.exe            00000000039D4EB0  Unknown               Unknown  Unknown
 27: fv3.exe            0000000002D37651  gfs_typedefs_mp_c        4465  GFS_typedefs.F90
```

**Comment by haiqinli** [comment]
> @haiqinli I see some conus13km cases crash.
> 
> ```
>  27: forrtl: severe (19): invalid reference to variable in NAMELIST input, unit -5, file Internal Formatted NML Read, line -1, p
> osition 20
>  27: Image              PC                Routine            Line        Source
>  27: fv3.exe            000000000399ADC8  Unknown               Unknown  Unknown
>  27: fv3.exe            00000000039D6ED7  Unknown               Unknown  Unknown
>  27: fv3.exe            00000000039D4EB0  Unknown               Unknown  Unknown
>  27: fv3.exe            0000000002D37651  gfs_typedefs_mp_c        4465  GFS_typedefs.F90
> ```

@jkbk2004 An input namelist variable of "smoke_forecast" is changed to "hwp_method". Or we can just simply remove "smoke_forecast" from the input.nml for regression test. Thanks.

**Comment by MatthewPyle-NOAA** [comment]
@jkbk2004 Thanks for updating the namelist template - looks good for me on WCOSS2 now.

**Comment by jkbk2004** [comment]
@MatthewPyle-NOAA We have Derecho resource issue. As Hera/Hercules/Wcoss2 tests are done, we can start merging this PR. We can merge physics pr https://github.com/ufs-community/ccpp-physics/pull/211

**Comment by jkbk2004** [comment]
@MatthewPyle-NOAA you can merge this pr with cleanup when you squash merge and add commit message.

File: tests/rt.sh [hunk 1]
```diff
@@ -381,7 +381,7 @@ elif [[ $MACHINE_ID = orion ]]; then
 
 elif [[ $MACHINE_ID = hercules ]]; then
 
-  module load contrib rocoto/1.3.5
+  module load contrib rocoto
```
Comment (grantfirl) [inline tests/rt.sh:384]: @MatthewPyle-NOAA This is what I'm talking about. Does this need to go to the develop branch?

**Pull request merged**